### ISSUE TITLE
[BUG] Content Security Policy blocks inline styles

### ISSUE THREAD

**Issue by sylvaindubus** [framer/motion#1727]
Hello,

I'm currently working on a project where we have a strict CSP which prevents inline execution of styles. This is problematic here as this library relies on style attributes (`style="opacity: 1; transform: none;"` for instance). Unfortunately, I couldn't find a workaround, like with a `nonce` attribute which we already use for other libraries. 

Thanks for any help.

Regards,

**Comment by mattgperry** [comment]
Unfortunately that functionality is core to all JavaScript animation libraries. I think to fix you'd have to stick to CSS animations and transitions.

**Comment by ianwensink** [comment]
Hi @mattgperry. Is this really the only acceptable answer? Is there no way to have a workaround for this?

Has no one ever used a strict CSP ánd Framer motion (or any other animation library) at the same time?

**Comment by bricejar** [comment]
@mattgperry @ianwensink @sylvaindubus  Hi ! Tell me if I need to open another issue. There might be a way to satisfy a stricter Content Security Policy.  

Globally allowind inline styles on a web app is a security vulnerability. If an attacker finds a way to display html with a malicious inline style in your website they could do a lot of harm. 
With CSP, you can allow dynamic style tags with a randomly generated nonce. So instead of showing the styles inline, you could create a class and generate a <style> to style this class. I assume this could have a performance impact but that would be the best mitigation strategy. 

Unfortunately right now there is no way to hook into the framer-motion machine to prevent inline styles from being added and transform them into a class. Just food for thought !

**Comment by fdev** [comment]
Not all methods of setting inline styles are blocked by `style-src self`. Basically, using `element.style.cssProp = value` from javascript doesn't violate CSP.

See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/style-src for examples of what is and what isn't blocked.

**Comment by jv4pres** [comment]
as @fdev points out, there is no issue once the page is loaded -- Framer Motion is already fully compatible with a strict CSP today *on the client side*, because setting styles via script is supported by all security policies. this issue is limited to server-side rendering; if the server generates an inline `style` attribute, the client will raise a security error.

so we just need a way to prevent rendering the `style` attribute on the server. the fix might be simple: Motion [already supports](https://www.framer.com/motion/component/##server-side-rendering) a `false` value for the `initial` prop which does 90% of what's needed. perhaps another value for `initial` could be used to indicate "the component has already been styled properly for its initial state, so please don't render any style attribute at all".

**Comment by Laith-PWC** [comment]
@mattgperry @ianwensink @sylvaindubus Hi!
I am trying to implement the nonce attribute, but the initial and animate attributes are generating inline styling, which renders the nonce useless. Am I missing something here?

I am using version: 11.18.2

Example implementation:
```
<MotionConfig nonce="123">
              <motion.div
                initial={{ height: "112px" }}
                animate={{
                  height: showAllItems
                    ? "auto" : "9vh"}`,
                }}
                transition={{ duration: 0.3 }}
                className=`flex flex-col gap-1`
                ref={scrollRef}
              >
```

the generated HTML div tag
`<div class="flex flex-col gap-1" style="height: 9vh;">`
